### PR TITLE
ci: various improvements

### PR DESCRIPTION
**Pull request by katexochen**
### Checklist

- [x] Add labels (e.g., for changelog category)

### CODE REVIEW

File: .github/workflows/e2e-test-release.yml [hunk 1]
```diff
@@ -37,68 +37,124 @@ jobs:
...
+          - test: "verify"
+            provider: "azure"
+            kubernetes-version: "v1.26"
+            runner: "ubuntu-22.04"
+          - test: "verify"
             provider: "aws"
-          - test: "perf-bench"
+            kubernetes-version: "v1.26"
+
+            runner: "ubuntu-22.04"
```
Comment (3u13r) [inline .github/workflows/e2e-test-release.yml:98]:
```suggestion
            runner: "ubuntu-22.04"

```

**Review by 3u13r — APPROVED** [review]
LGTM

File: .github/workflows/e2e-test-weekly.yml [hunk 2]
```diff
@@ -45,66 +45,130 @@ jobs:
...
-        kubernetes-version: ["v1.24", "v1.25", "v1.26"]
-        refStream: ["ref/main/stream/debug/?", "ref/release/stream/stable/?"]
-        exclude:
-          # Verify test runs only on latest kubernetes-version by default.
-          # Exclude it for main-debug as it needs to run on 1.25 / release-stable.
-          - test: "verify"
-            kubernetes-version: "v1.24"
-          - refStream: "ref/main/stream/debug/?"
-            test: "verify"
+        include:
```
Comment (msanft) [inline .github/workflows/e2e-test-weekly.yml:48]: good idea, way better than the exclude matrix

**Review by msanft — APPROVED** [review]
LGTM

**Comment by msanft** [comment]
One minor question for understanding:

Whats the benefit of using the `tee -a` pipe over `>>`? I find the latter more descriptive

**Pull request merged**